### PR TITLE
fix: navigation issue in the Iframe widget due to src change

### LINKED ISSUE #15585: [Bug]-[64]:iFrame - On setting onSrcDocChanged event to navigate to a different page, user is unable to return to the original page where the iFrame is present.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

https://www.loom.com/share/c99ecf443f2944e1aae6082ce067a355

### Steps To Reproduce

1. Drop an iFrame widget and add in the srcDoc details
2. Set the onClick event to navigate to a different page (same window) onSrcDocChanged
3. Change the content of the SrcDoc
4.  User gets navigated to the set page (page 1 in this case) but is unable to come back to page 4 (original page)

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Fixed the navigation issue in the Iframe widget due to the src change.

Fixes #15585 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/IframeWidget/component/index.tsx [hunk 1]
```diff
@@ -115,8 +116,8 @@ function IframeComponent(props: IframeComponentProps) {
   }, [source]);
 
   useEffect(() => {
-    if (isFirstRender.current) {
-      isFirstRender.current = false;
+    if (isFirstSrcDocRender.current) {
```
Comment (keyurparalkar) [inline app/client/src/widgets/IframeWidget/component/index.tsx:119]: Is this the condition that prevents the onSrcDocChange to execute on first render? @techbhavin

File: app/client/src/widgets/IframeWidget/component/index.tsx [hunk 2]
```diff
@@ -102,8 +103,8 @@ function IframeComponent(props: IframeComponentProps) {
   }, []);
 
   useEffect(() => {
-    if (isFirstRender.current) {
-      isFirstRender.current = false;
+    if (isFirstURLRender.current) {
```
Comment (keyurparalkar) [inline app/client/src/widgets/IframeWidget/component/index.tsx:106]: @techbhavin are we renaming the variable here?

**Review by keyurparalkar — CHANGES_REQUESTED** [review]
@techbhavin can we add cypress test cases here

File: app/client/src/widgets/IframeWidget/component/index.tsx (hunk 1 above)
Comment (techbhavin) [reply app/client/src/widgets/IframeWidget/component/index.tsx:119]: Yes @keyurparalkar

**Comment by laveena-en** [comment]
Tested and verified.

**Review by keyurparalkar — APPROVED** [review]
LGTM!

**Pull request merged**